templates/managed-namespaces-zones.tf

```diff
@@ -0,0 +1,128 @@
+{{- $allNamespaces := prepend (dict "name" "gsp-system" "ingress" (dict "enabled" true)) (datasource "config").namespaces }}
+{{- range $namespace := $allNamespaces }}
+{{- if (has $namespace "ingress") }}
+{{- if (has $namespace.ingress "enabled") }}
+{{- if $namespace.ingress.enabled }}
+
+resource "aws_route53_zone" "{{ $namespace.name }}" {
+  name          = "{{ $namespace.name }}.${var.cluster_domain}"
+  force_destroy = true
+}
+
+resource "aws_route53_record" "{{ $namespace.name }}-ns" {
+  zone_id  = module.gsp-domain.zone_id
+  name     = "{{ $namespace.name }}.${var.cluster_domain}"
+  type     = "NS"
+  ttl      = "30"
+
+  records = [
+    aws_route53_zone.{{ $namespace.name }}.name_servers[0],
+    aws_route53_zone.{{ $namespace.name }}.name_servers[1],
+    aws_route53_zone.{{ $namespace.name }}.name_servers[2],
+    aws_route53_zone.{{ $namespace.name }}.name_servers[3],
+  ]
+}
+
+data "aws_iam_policy_document" "trust_{{ $namespace.name }}_external_dns" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRoleWithWebIdentity"]
+
+    principals {
+      type        = "Federated"
+      identifiers = [module.k8s-cluster.oidc_provider_arn]
```

Comment:
I'm not 100% sure ... but should this be `module.gsp-cluster` not `k8s-cluster` (since we are in the deployer project not the gsp-cluster module) ....and then gsp-cluster module would need an `oidc_provider_arn` output (which I don't see)

Reply:
yeah, this makes sense to me. I'll push a change